upgrade_policy/quorum_upgrade/scripts/ts/src/transactions.ts

```diff
@@ -67,6 +73,66 @@ const proposeUpgrade = (txb: TransactionBlock, quorumUpgradeCapId: string, packa
 
 }
 
+/// Calls `quorum_upgrade_policy::propose_upgrade_v2`, `quorum_upgrade_policy::add_all_metadata` (optional), `quorum_upgrade_policy::share_upgrade_object`,  
+/// Calling this will publish the `ProposedUpgrade` shared object.
+const proposeUpgradeV2 = (txb: TransactionBlock, quorumUpgradeCapId: string, packagePath: string, metadata?: { [key: string]: Uint8Array } | null) => {
+    const { digest }  = getUpgradeDigest(packagePath);
+
+    const proposedUpgrade = txb.moveCall({
+        target: `${QUORUM_UPGRADE_PACKAGE_ID}::quorum_upgrade_policy::propose_upgrade_v2`,
+        arguments: [
+            txb.object(quorumUpgradeCapId),
+            txb.pure(digest, 'vector<u8>')
+        ]
+    });
+    
+    if (metadata) {
+        txb.moveCall({
+            target: `${QUORUM_UPGRADE_PACKAGE_ID}::quorum_upgrade_policy::add_metadata`,
+            arguments: [
+                txb.object(proposedUpgrade),
+                prepareMetadataVecMap(txb, metadata)
+            ]
+        });
+    }
+
+    txb.moveCall({
+        target: `${QUORUM_UPGRADE_PACKAGE_ID}::quorum_upgrade_policy::share_upgrade_object`,
+        arguments: [
+            txb.object(proposedUpgrade),
+        ]
+    });
+}
+
+/// Use the `ProposedUpgrade` object id to get the optional metadata
+const getMetadata = async (proposedUpgradeObjectId: string) => {
+    try {
+        const result = await client.call<{ data: [{objectId: string}] }>('suix_getDynamicFields', [proposedUpgradeObjectId]);
+
+        // Assuming the result structure correctly, extract the dynamic field ID
+        const dynamicFieldId = result.data[0].objectId;
+        
+        // Fetch the content associated with the dynamic field ID
+        const output = await client.call<{data: any}>('sui_getObject', [dynamicFieldId,{"showContent": true,}]);
+
+        const arr = output.data.content.fields.value.fields.contents;
```

Comment:
Wondering if there's a better way to do this

Reply:
There's no better way to do that!